tests/Model/BSONIteratorTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use MongoDB\Exception\UnexpectedValueException;
+use MongoDB\Model\BSONArray;
+use MongoDB\Model\BSONDocument;
+use MongoDB\Model\BSONIterator;
+use MongoDB\Tests\TestCase;
+use stdClass;
+
+class BSONIteratorTest extends TestCase
+{
+    /**
+     * @dataProvider provideTypeMapOptionsAndExpectedDocuments
+     */
+    public function testValidValues(array $typeMap, $binaryString, array $expectedDocuments)
+    {
+        $bsonIt = new BSONIterator($binaryString, $typeMap);
+
+        $this->assertSameDocuments($expectedDocuments, $bsonIt);
+    }
+
+    public function provideTypeMapOptionsAndExpectedDocuments()
+    {
+        return [
+            [
+                [],
```

Comment:
Let's make sure that one test passes `null` as the `typeMap` option, which will ensure code coverage of the constructor assigning `[]` in that case.